homeassistant/components/fritz/device_tracker.py

```diff
@@ -60,8 +60,6 @@ async def async_get_scanner(hass: HomeAssistant, config: ConfigType) -> None:
         "Loading Fritz via scanner setup is now deprecated"
     )
 
-    return None
```

Comment:
The problem I see is in cases like this, a `get_scanner` method that now returns nothing. Or a function that can return `X | None`, we should be able to just do `return None`. Relying on a pylint disable rule in such cases doesn't seem like a step forward.

If a return value of a function is never checked by a caller, we can omit `return None`, but I don't think that it's possible to detect / very common that it warrants a rule.

Reply:
I think `get_scanner` doesn't have a good name here, it does indeed return nothing in any case:

````python3
async def async_get_scanner(hass: HomeAssistant, config: ConfigType) -> None:
```

This plugin only is only intended to catch cases when function is annotated with `-> None`. If function returns `X | None` then yes, `return None` should be used, this plugin won't complain about it.

Reply:
`async_get_scanner` should be annotated as `DeviceScanner | None` actually.

Reply:
In that case it is catching some cases, however it's catching mainly cases where the return type was bad – not the return value. And the way the warning is specified, the user will be inclined to remove the return statement and not fix the return type (as happened in this case here).

Reply:
Right, the fix is not correct here, return type should be changed instead.

I agree, warning message should be corrected to include such case.

Reply:
Alright, in that case I agree it's an improvement, but we should make sure our warning message is clear that the issue can be with the return type OR the return value on the line that the issue was flagged.

Reply:
@KapJI please clarify the PR description with your comment and example about optional return values. The current description doesn't explain this which is misleading.